Fix `bitmer_to_bytes` and add quickcheck.

theJasonFan opened the pull request:
Fixed bitmer_to_bytes to match endianness in tests.

Also added quickcheck back to tests.

rob-p commented:
Thanks @theJasonFan!  Can you explain the bug a little bit?  Also, I'm unfamiliar with quickcheck.  What's the difference / benefit compared to normal unit tests?

theJasonFan commented:
### The `kmer::tests::kmer_prefix` test was failing
This was because `bitmer_to_bytes` wrote the highest order bits in the `u64` representing a kmer in the lowest indexed positions of the `Vec<u8>`. As I understand, the correct implementation  should instead write the lowest order bits in the lowest indexed positions, as kmers (read as strings) should be encoded from low-to-high order bits. I fixed `bitmer_to_bytes` reflect this, making that `kmer::tests::kmer_prefix`pass.

### On `quickcheck`
`quickcheck` lets you test functions with randomized input, it was originally developed for Haskell and have since been implemented in Rust. There were several unused functions in tests that had a commented out `#[quickcheck]` annotation -- I  assumed that these were removed at some point. I know that my original naive kmer implementation that `crate::naieve_impl` resembles (is based off of?) had used quickcheck.

I just added `quickcheck` back to dev dependencies and made sure that no tests were warned to be unused.
